### PR TITLE
build.h: add a bunch of missing features strings

### PR DESCRIPTION
**Pull request by poettering**
Let's make "systemctl --version" more useful for detecting when
build-time features/deps are enabled and which ones arent't again.

### CODE REVIEW

**Comment by yuwata** [comment]
Build fails.

**Comment by yuwata** [comment]
BTW, if possible, I think it is better these strings are automatically generated by meson.

**Comment by poettering** [comment]
Hmm, we could probably move it there, but we need to redact the string manually either way: we want to list many, but not all features there

**Comment by poettering** [comment]
Force pushed a new version btw, fixing the commit hopefully. But still generating the string in C code, not meson.

**Comment by yuwata** [comment]
LGTM.

**Comment by keszybz** [comment]
LGTM too.

**Comment by keszybz** [comment]
On closer look, it's not quite right.
I'll push my patch on top to rework things a bit.

**Comment by yuwata** [comment]
Not a matter, but it is better if compressors are also sorted.

**Comment by keszybz** [comment]
Indeed. Updated.

File: src/basic/build.c [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        " -ZSTD"
+#endif
+
+        /* backwards compat integration */
+
+#if HAVE_XKBCOMMON
+        " +XKBCOMMON"
+#else
+        " -XKBCOMMON"
+#endif
```
Comment (poettering) [inline src/basic/build.c:201]: this isn't really backwards compat...

**Comment by poettering** [comment]
(one of those days we should add some color to the feature output, and grey out the unsupported features... ;-))

File: src/basic/build.c (hunk 1 above)
Comment (keszybz) [reply src/basic/build.c:201]: Kind-of:

Adam Jackson (@ajax): "on abandoning the X server"                                                       
https://ajaxnwnk.blogspot.com/2020/10/on-abandoning-x-server.html                                        
                                                                                                         
Also Daniel Vetter (Intel) is on record that the xserver might be                                        
considered "abandonware".                                                                                
https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/533#note_670522

Comment (poettering) [reply src/basic/build.c:201]: yeah, but xkbcommon is not x11. it's also used in a wayland world really. it's the way how kbd mappings are done today.

right, @whot?

Comment (keszybz) [reply src/basic/build.c:201]: It's just a comment in the file that nobody looks at. But you're right, xkbcommon is not "backwards compat", it doesn't even have much to to with X nowadays. I updated the comment.

**Comment by yuwata** [comment]
LGTM.

**Comment by poettering** [comment]
lgtm too

File: src/basic/build.c (hunk 1 above)
Comment (whot) [reply src/basic/build.c:201]: > xkbcommon is not x11

yep, that's certainly correct. Anything Waylandy uses libxkbcommon but the x server itself still uses its own parser (and `xkbcomp`) and we can't easily switch it. Aside from the XKB structural layout, there isn't anything X11-specific in libxkbcommon itself.

**Pull request merged**